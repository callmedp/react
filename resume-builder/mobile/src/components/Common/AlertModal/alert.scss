@import '../../../styles/variables/vars.scss';
@import '../../../styles/variables/mixins.scss';

.modal-image{
    text-align: center;
    margin-top: 30px;
}

.alertModal {
    display: flex;
    justify-content: center;
    height: 100%;
    align-items: center;

    &__wrap {
        @include border-radius(.8rem);
        background-color: $white;
        margin: 1.5rem;
        padding: 3rem;

        &--title {
            color: $black;
            font-size: 2rem;
        }
        
        &--content {
            display: flex;
            flex-direction: column;
            align-items: center; 
            font-size: 1.4rem;
            color: $grey-666;
        }
        
        &__btn-wrap {
            display: flex;
            flex-direction: column;
            align-items: center;

            span {
                text-align: center;
                margin: 5px 0;
            }
        }
    }
}