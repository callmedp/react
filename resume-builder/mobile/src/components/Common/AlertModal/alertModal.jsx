import React,{Component} from 'react';
import Modal from 'react-modal';
import './alert.scss'


Modal.setAppElement(document.getElementById('react-app'));

export default class AlertModal extends Component{

    constructor(props){
        super(props);
        this.staticUrl = window && window.config && window.config.staticUrl || '/media/static/';
        this.goToNewLink = this.goToNewLink.bind(this);

    }

    goToNewLink(){
        const {history,link,closeModal} = this.props;
        history.push(`/resume-builder/edit/?type=${link}`)
        closeModal()

    }
    render(){
        const {modal_status,closeModal,newUser,generateResumeModal} = this.props;
        return(
            <Modal 
                isOpen={modal_status} 
                contentLabel="onRequestClose Preview"
                onRequestClose={this.handleCloseModal}
                className="alertModal"
                overlayClassName="Overlay">
                {newUser ?
                    <div className="alertModal__wrap">
                        <i className="alertModal__wrap--alert m-auto"></i>
                        <div className="alertModal__wrap--title text-center mb-15">Please save your profile info to continue</div>
                        {/* <div className="alertModal__wrap--content text-center mb-15">
                            <p>Some information may be lost as required fields are not filled.</p>
                        </div> */}
                        <div className="alertModal__wrap__btn-wrap">
                            {/* <span className="btn btn-sm btn__primary btn__round w-150" onClick={this.goToNewLink}>Yes, change it!</span> */}
                            <span className="btn btn-sm btn--outline btn__round w-150" onClick={closeModal}>Ok</span>
                        </div>
                    </div>:
                    generateResumeModal ?
                    <div className="alertModal__wrap">
                        <div className="alertModal__wrap--title text-center mb-15">Generating Resume</div>
                        <div className="alertModal__wrap--content text-center mb-15">
                            <p>Please wait a for some time till we generate your resume.</p>
                        </div>
                        <div className="logo-center">
                            <img src={`${this.staticUrl}react/assets/images/mobile/blue-loader.png`}/>
                        </div> 
                    </div>:
                    <div className="alertModal__wrap">
                        <i className="alertModal__wrap--alert m-auto"></i>
                        <div className="alertModal__wrap--title text-center mb-15">Are you sure?</div>
                        <div className="alertModal__wrap--content text-center mb-15">
                            <p>Some information may be lost as required fields are not filled.</p>
                        </div>
                        <div className="alertModal__wrap__btn-wrap">
                            <span className="btn btn-sm btn__primary btn__round w-150" onClick={this.goToNewLink}>Yes, change it!</span>
                            <span className="btn btn-sm btn--outline btn__round w-150" onClick={closeModal}>Cancel</span>
                        </div>
                    </div>
                }
            </Modal>
        )
    }
}