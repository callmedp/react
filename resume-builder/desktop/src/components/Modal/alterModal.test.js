import { shallow} from 'enzyme';
import React from 'react';
import AlertModal  from './alertModal';


describe('AlertModal', () => {

    it('Alert modal  debug case ', () => {  
        const props = {
            ui: {
                alertModal : true,
                generateResumeModal: false
            },
            nextLink: null,
            newUser: false,
            isPreview: false
        },
         ModalWrapperComponent = shallow(<AlertModal {...props} />);
        expect(ModalWrapperComponent).toMatchSnapshot();
    });
})

