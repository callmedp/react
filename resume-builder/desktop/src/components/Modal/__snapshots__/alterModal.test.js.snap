// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertModal Alert Modal 1`] = `
<Fragment>
  <div
    className="pr alert-modal"
  >
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      closeTimeoutMS={0}
      contentLabel="Example Modal"
      isOpen={true}
      onRequestClose={[Function]}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={false}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {
            "bottom": "0",
            "height": "280px",
            "left": "0",
            "margin": "auto",
            "right": "0",
            "top": "10%",
            "width": "450px",
          },
        }
      }
    >
      <div
        className="pr"
      >
        <div
          className="alert-modal"
        >
          <span
            className="icon-alert"
          />
          <strong>
            Are you sure?
          </strong>
          <p>
            Some information may be lost as required fields are not filled.
          </p>
          <div
            className="flex-container"
          >
            <button
              className="orange-button mr-10"
              onClick={[Function]}
            >
              Yes, change it.
            </button>
            <button
              className="blue-button"
              onClick={[Function]}
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
    </Modal>
  </div>
</Fragment>
`;

exports[`AlertModal Alert modal  debug case  1`] = `
<Fragment>
  <div
    className="pr alert-modal"
  >
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      closeTimeoutMS={0}
      contentLabel="Example Modal"
      isOpen={true}
      onRequestClose={[Function]}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={false}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {
            "bottom": "0",
            "height": "280px",
            "left": "0",
            "margin": "auto",
            "right": "0",
            "top": "10%",
            "width": "450px",
          },
        }
      }
    >
      <div
        className="pr"
      >
        <div
          className="alert-modal"
        >
          <span
            className="icon-alert"
          />
          <strong>
            Are you sure?
          </strong>
          <p>
            Some information may be lost as required fields are not filled.
          </p>
          <div
            className="flex-container"
          >
            <button
              className="orange-button mr-10"
              onClick={[Function]}
            >
              Yes, change it.
            </button>
            <button
              className="blue-button"
              onClick={[Function]}
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
    </Modal>
  </div>
</Fragment>
`;
